aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -316,10 +313,12 @@ fn native_entry_function_payload_internal(
                 TRANSACTION_CONTEXT_ENTRY_FUNCTION_PAYLOAD_PER_BYTE_IN_STR
                     * NumBytes::new(num_bytes as u64),
             )?;
-            let payload = create_entry_function_payload(entry_function_payload);
-            Ok(smallvec![create_option_some_value(payload)])
+            let payload = create_entry_function_payload(entry_function_payload)?;
+            // SAFETY: payload is a struct, so it is ok to create a vector out of it. Same applies
```

Comment:
the unchecked doc and messages say check for vec is that types are the same. If so, the safety here (and next) is that we are ever adding 1 or 0 elements (some struct / none), isn't it?

Reply:
Good point, types are checked otherwise: it is 1/0 which guarantee safety - adjusted the comment.